builtInFunctions/esdtNFTCreate.go

```diff
@@ -112,17 +119,42 @@ func (e *esdtNFTCreate) ProcessBuiltinFunction(
 	if err != nil {
 		return nil, err
 	}
-	if len(vmInput.Arguments) < 7 {
+
+	minNumOfArgs := 7
+	if vmInput.CallType == vm.ExecOnDestByCaller {
```

Comment:
we need to condition the usage of the vm.ExecOnDestByCaller type against an epoch.

Reply:
it is conditioned in Arwen.